### PR TITLE
WIP import script bbcode fixes in Fluxbb importer + some fixes to all importers

### PR DESCRIPTION
**Pull request by harry-wood**
Not really work in progress any more, but there's quite a lot here so I think I should break this into smaller more manageable PRs for review. I have already done so with the very first step: https://github.com/discourse/discourse/pull/18893  If we can get that merged, then I will carve off more of this big beast into separate PRs.

We're looking to run a big FluxBB import (https://forum.openstreetmap.org/ -> https://community.openstreetmap.org/) and with the code in this branch we're hopefully ready to go for that purpose (although we're doing large scale testing still, so some further changes may arise)

At the moment there's currently a cheeky commit here to do a Gemfile path swap, to include [some changes I've been making to the ruby-bbcode-to-md gem](https://github.com/nlalonde/ruby-bbcode-to-md/pull/6). If I can get those merged then, no need for the Gemfile path swap.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -257,7 +257,7 @@ if ENV["IMPORT"] == "1"
 
   # NOTE: in import mode the version of sqlite can matter a lot, so we stick it to a specific one
   gem 'sqlite3', '~> 1.3', '>= 1.3.13'
-  gem 'ruby-bbcode-to-md', git: 'https://github.com/nlalonde/ruby-bbcode-to-md'
+  gem 'ruby-bbcode-to-md', git: 'https://github.com/harry-wood/ruby-bbcode-to-md', branch: 'fluxbb-tests-and-fixes'
```
Comment (gschlager) [inline Gemfile:260]: Could you please create a PR with your changes at https://github.com/nlalonde/ruby-bbcode-to-md?

Comment (harry-wood) [reply Gemfile:260]: Indeed I have done so (see description above). So I'm hoping this Gemfile change would eventually not be necessary, although... I don't know if nlalonde is intending to merge any PRs there and maintain that fork of the gem. D'you know what the status of that is?

Comment (gschlager) [reply Gemfile:260]: He's on our team, so I'm sure he will merge the PR when it's ready for review. Just mention me here when your PRs are ready for review and I'll get everything sorted.

**Comment by nattsw** [comment]
Hey there, we're tackling older PRs and will close PRs with no follow up. Please open a new one when you're ready.

**Pull request closed without merging**